tls/s2n_server_hello.c

```diff
@@ -107,6 +71,53 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
         GUARD(s2n_server_extensions_recv(conn, &extensions));
     }
 
+    if (conn->server_protocol_version >= S2N_TLS13) {
+        /* Check echoed session ID matches */
+        if(session_id_len != conn->session_id_len || !memcmp(session_id, conn->session_id, session_id_len)){
+            GUARD(s2n_queue_reader_illegal_parameter_alert(conn));
+        }
+        conn->actual_protocol_version = conn->server_protocol_version;
```

Comment:
Why set actual to server here, instead of in the extension?

Reply:
i set it here because for protocol_version < TLS13, it was set here with either session resumption or not. TLS 1.3 doesn't do sessions the same way but it made it easy to see the side by side comparison. That being said, it does not need to be here and i can put it in the extension if you think that's more clear.